lib/src/lifecycle_module.dart

```diff
@@ -205,12 +212,13 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
   void specifyFirstUsefulState({
     Map<String, dynamic> tags: const {},
     List<Reference> references: const [],
-  }) =>
-      specifyStartupTiming(
-        StartupTimingType.firstUseful,
-        tags: tags,
-        references: references,
-      );
+    DateTime trueStartTime,
```

Comment:
Definitely open to renaming this. Essentially this allows the module to specify that the "true start time" (i.e, the actual wait a consumer perceives) isn't necessarily the moment that the module's `load` function is called.

Reply:
Could you provide an example where this would result in a meaningful difference? I'm concerned with adding an option for overriding the start time because we want to provide as much of this profiling information out of the box as possible. If we don't have a well-defined, consistent start time, the metrics between experiences may vary considerably based on how `trueStartTime` is used.

Reply:
Yeah so in a sox test form we load up the first Phase module and overview module. The user is on the overview for an indeterminate amount of time, but the phase module's onLoad has already been called. It could be 5 minutes before they actually open the phase, but the only time we care about is the time from them clicking that tab -> first render of the components / modules in that tab.

Reply:
Based on that use-case, I would ask:
1. Why is the module being loaded if it's not being used
2. This timing is defined as the span between load and interactive; as opposed to changing this definition, would a custom tracing span to capture the time between "phase opened" and "interactive" be a better fit?

Reply:
1. It's preloading part of the data so that it's partially available when the user switches tabs, instead of cold starting on every tab switch (this significantly improved load times for us some time ago).
 
2. You're right. Perhaps we could modify that definition because it assumes a relatively rigid use of the module? Generally what we're trying to track is perceived load time by the user, not necessarily whatever the component lifecycle ends up being. The onLoad span tracks the lifecylce, the firstUseful should be more ad-hoc and defined by the module implementation IMO.

I think we could probably adjust the definition to:
```
  /// Creates a span with `globalTracer` to track the perceived load time of this module.
  ///
  /// This span is intended to represent the time it takes for the module to
  /// finish asynchronously loading any necessary data and entering a state which
  /// is ready for user interaction.
```

Thoughts?

Reply:
After re-reading your second question, I think you're right. We could just define a manual span that captures what we want to capture. That would remove the need for us to overwrite this start time here. 

We would still want the isFirstUseful future and "isTraced" bool property though. Does that sound good?

Reply:
I actually think your definition makes a lot more sense; we defined these without having very clear use-cases, so it makes sense to change them to match how users want to use them.

I'd be open to that definition changing, but I think it'd be nice for the new "start" point to be defined via an API closer to the creation of a span, as opposed to passing in a time.

For example, instead of :
```dart
var startTime = new DateTime.now();
// ...
specifyFirstUsefulState(startTime: startTime);
```
```dart
final firstUsefulSpan = startFirstUsefulStateSpan();
//
firstUsefulSpan.finish();
```

This has the advantage of recording these spans as marks/measures, and keeps managing of the start/end points of the span more consistent with other tracing APIs.

---

> We would still want the isFirstUseful future and "isTraced" bool property though. Does that sound good?

👍

Reply:
That would be awesome, I would prefer that as well because the dart tracing console doesn't show spans with start times that were injected. So they just end up being 1ms spans (at least visually, but in on our tracing dashboard).

The only issue is that some of the module's children are preloaded during the onLoad. We want both those children that were pre-loaded (so loaded before the startFirstUsefulStateSpan()) and the ones called during the first render cycle to share the same parent context.

The current method allows both the pre-loaded and loaded-on-demand spans to share a parent context. Does that make sense? Any thoughts for a way to fix that issue?

Reply:
As per offline discussion, this has been addressed in https://github.com/Workiva/w_module/pull/143/commits/9520d8704d36f43259b5c89efc29bc3171328443